bundles/org.openhab.binding.tibber/src/main/java/org/openhab/binding/tibber/internal/handler/TibberHandler.java

```diff
@@ -297,7 +283,34 @@ public void updateRequest() throws IOException {
         getURLInput(BASE_URL);
         if ("true".equals(rtEnabled) && !isConnected()) {
             logger.debug("Attempting to reopen Websocket connection");
-            open();
+            getSubscriptionUrl();
+
+            if (subscriptionURL == null || subscriptionURL.isBlank()) {
+                logger.debug("Unexpected subscription result from the server");
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                        "Unexpected subscription result from the server");
+            } else {
+                logger.debug("Reconnecting Subscription to: {}", subscriptionURL);
+                open();
+            }
+        }
+    }
+
+    public void getSubscriptionUrl() throws IOException {
```

Comment:
It looks like this method could be private?

Reply:
Method made private.